### PR TITLE
Image seeds

### LINKED ISSUE #6: Image Seeds
#### As a developer I want at least 20 images to be created when I setup the database.

__Story__:

Working on this application has been a lot of fun, right? You are fast
becoming a git expert, and an expert at working with a deployment
pipeline. Speaking of development and deployment pipelines, is it not pretty
annoying that everytime a new deployment is created it starts out with an empty
database? It seems like everytime a review application is created we need to
manually add a handful of images in order to manually test the
application. Would it not be great if each deployment was created with a
handful of existing images?

__Acceptance criteria__:
- [ ] After running `rake db:setup` at least 20 unique images are shown in the
  homepage.
- [ ] When a new app is deployed on heroku (e.g., every review app), that app
  automatically has at least 20 unique images on its homepage.

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by HongquanYu**
Closes #6 

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [x] I have replaced `X` above with the associated issue number.
- [x] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [x] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by HongquanYu** [comment]
[Image seeds feature](https://trello.com/c/kHW0Ab6b/372-image-seeds-feature)

File: .rubocop.yml [hunk 1]
```diff
@@ -19,7 +19,10 @@ Layout/MultilineMethodCallIndentation:
   EnforcedStyle: indented
 
 Metrics/LineLength:
-  Max: 112  # Maximum line length from `rails new`
+  Max: 200  # Maximum line length from `rails new`
```
Comment (abineetds) [inline .rubocop.yml]: Doing this isn't usually great practice. The purpose of rubocop is to encourage good code style. You should wrap lines or split chained lines if they're too long. Same goes for class length. Length metrics aren't super enforced but we try to be as clean as we can wherever possible.

File: .rubocop.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ Layout/MultilineMethodCallIndentation:
   EnforcedStyle: indented
 
 Metrics/LineLength:
-  Max: 200  # Maximum line length from `rails new`
+  Max: 400  # Maximum line length from `rails new`
```
Comment (liyiran) [inline .rubocop.yml]: we can add a line annotation to suppress the error rather than change the global configuration.

Comment (HongquanYu) [reply .rubocop.yml]: Sounds good, changed.

File: db/seeds.rb [hunk 3]
```diff
@@ -5,3 +5,28 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
```
Comment (abineetds) [inline db/seeds.rb]: You want to get rid of the default comments here, but other than that this looks good.

File: db/seeds.rb [hunk 4]
```diff
@@ -5,3 +5,28 @@
...
+
+Image.create!(
```
Comment (liyiran) [inline db/seeds.rb]: we can also use a loop to insert the images if the images themselves are not so important. `http://placehold.it/120x120&text=image#{i}` supports image place holder, e.g. image1, image2, etc.

File: .rubocop.yml [hunk 5]
```diff
@@ -7,6 +7,7 @@ AllCops:
     - 'node_modules/**/*'
     - 'bin/*'
     - 'config/initializers/simple_form.rb'
+    - 'db/seeds.rb'
```
Comment (liyiran) [inline .rubocop.yml]: Excluding the whole file might not be a good idea. Adding `# rubocop:disable Metrics/ClassLength` to the corresponding lines in `seeds.rb` is better.

File: .rubocop.yml [hunk 6]
```diff
@@ -19,7 +20,7 @@ Layout/MultilineMethodCallIndentation:
   EnforcedStyle: indented
 
 Metrics/LineLength:
-  Max: 200  # Maximum line length from `rails new`
+  Max: 110  # Maximum line length from `rails new`
```
Comment (liyiran) [inline .rubocop.yml]: Is there any reason for changing the max length from 200 to 110?

File: .rubocop.yml (hunk 1 above)
Comment (HongquanYu) [reply .rubocop.yml]: Changed

File: .rubocop.yml (hunk 5 above)
Comment (HongquanYu) [reply .rubocop.yml]: Done with that

File: .rubocop.yml (hunk 6 above)
Comment (HongquanYu) [reply .rubocop.yml]: Changed it back to original value

File: db/seeds.rb (hunk 3 above)
Comment (HongquanYu) [reply db/seeds.rb]: done

**Comment by liyiran** [comment]
just a comment: the branch name and pull request name should contain `vm` or author's name, e.g. `HY-image-seeds`

**Pull request merged**